packages/itwin/grouping-mapping-widget/src/test/GroupingMappingCustomUI.test.tsx

```diff
@@ -0,0 +1,195 @@
+  description: "mOcKmApPiNgDeScRiPtIoN1",
+  createdBy: faker.random.alpha(),
+  createdOn: faker.date.past().toDateString(),
+  modifiedBy: faker.random.alpha(),
+  modifiedOn: faker.date.past().toDateString(),
+  extractionEnabled:false,
+  _links: {
+    imodel: {
+      href: "",
+    },
+  },
+};
+
+const groupsFactory = (): GroupCollection => ({
+  groups: Array.from(
+    { length: faker.datatype.number({ min: 3, max: 5 }) },
+    (_, index) => ({
+      id: `${faker.datatype.uuid()}`,
+      groupName: `mOcKgRoUp${index}`,
+      description: `mOcKgRoUpDeScRiPtIoN${index}`,
+      query: `mOcKgRoUpQuErY${index}`,
+      deleted: false,
+      _links: {
+        imodel: {
+          href: "",
+        },
+        mapping: {
+          href: "",
+        },
+      },
+    })
+  ),
+  _links: {
+    next: undefined,
+    self: {
+      href: "",
+    },
+  },
+});
+
+const connectionMock = moq.Mock.ofType<IModelConnection>();
+const selectionManagerMock = moq.Mock.ofType<SelectionManager>();
+const selectionScopesManagerMock = moq.Mock.ofType<SelectionScopesManager>();
+
+jest.mock("@itwin/appui-react", () => ({
+  ...jest.requireActual("@itwin/appui-react"),
+  useActiveIModelConnection: () => connectionMock.object,
+}));
+
+const server = setupServer();
+
+beforeAll(async () => {
+  // This is required by the i18n module within iTwin.js
+  (global as any).XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest; // eslint-disable-line @typescript-eslint/no-var-requires
+  await NoRenderApp.startup();
+  await Presentation.initialize();
+  const selectionSet = moq.Mock.ofType<SelectionSet>();
+  const onChanged = moq.Mock.ofType<BeEvent<(ev: SelectionSetEvent) => void>>();
+  selectionSet.setup((x) => x.elements).returns(() => new Set([]));
+  selectionSet.setup((x) => x.onChanged).returns(() => onChanged.object);
+  connectionMock
+    .setup((x) => x.selectionSet)
+    .returns(() => selectionSet.object);
+  connectionMock.setup((x) => x.iModelId).returns(() => mockIModelId);
+  connectionMock.setup((x) => x.iTwinId).returns(() => mockITwinId);
+
+  selectionManagerMock
+    .setup((x) => x.selectionChange)
+    .returns(() => new SelectionChangeEvent());
+
+  selectionScopesManagerMock
+    .setup(async (x) => x.getSelectionScopes(connectionMock.object))
+    .returns(async () => []);
+  selectionManagerMock
+    .setup((x) => x.scopes)
+    .returns(() => selectionScopesManagerMock.object);
+
+  Presentation.setSelectionManager(selectionManagerMock.object);
+  await TestUtils.initializeUiFramework(connectionMock.object);
+  server.listen({
+    onUnhandledRequest: "warn",
+  });
+  server.printHandlers();
+});
+
+afterAll(() => {
+  server.close();
+});
+
+afterEach(() => {
+  server.resetHandlers();
+});
+
+describe("Groupings View with default UIs", () => {
+  it("List all groups and click on add group and edit group buttons", async () => {
+    // Arange
+    const mockGroups = groupsFactory();
+    server.use(
```

Comment:
You should be mocking the insights client and not checking each request. What if the internals of insights client would change? Then you would need to update all of the tests.